app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,46 @@
 package com.lightricks.feedexercise.data
 
+import com.lightricks.feedexercise.database.FeedItemEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.FeedApiService.Companion.THUMBNAILS_URL
+import com.lightricks.feedexercise.network.TemplateMetadata
+import com.lightricks.feedexercise.network.TemplatesMetadata
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.map
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
-}
\ No newline at end of file
+class FeedRepository(
+    private val feedApiService: FeedApiService = FeedApiService.create()) {
+    private val cachedFeedItems = MutableStateFlow<List<FeedItemEntity>>(emptyList())
+
+    fun feedItemsFlow(): Flow<List<FeedItem>> = cachedFeedItems.map { entities ->
+        entities.map { entity ->
+            FeedItem(
+                entity.id,
+                entity.thumbnailUrl,
+                entity.isPremium)
+        }
+    }
+
+    suspend fun refreshFeedItems() {
+        val templatesMetadata: TemplatesMetadata = feedApiService.getTemplatesMetadata()
+        val refreshedFeedItemEntities = templatesMetadata.templates.map { templateMetadata ->
+            feedItemEntityFromTemplateMetadata(templateMetadata)
+        }
+        cachedFeedItems.value = refreshedFeedItemEntities
```

Comment:
This should run on an IO thread so please ensure that by making a context switch if needed.

Reply:
The only place where this is used is in line 43, where I added a context switch to dispatchers.IO

Reply:
I see you make the CS on the call site, as best practice is better to put it on the implementation since the ones that better knows where it is the most appropriate to run the code is the author of the code not the client.